channel.go

```diff
@@ -50,29 +51,39 @@ type channel struct {
 	cancelStream    context.CancelFunc
 	responseRouters map[uint64]responseRouter
 	responseMut     sync.Mutex
+	connEstablished bool
 }
 
 func newChannel(n *RawNode) *channel {
 	return &channel{
 		sendQ:           make(chan request, n.mgr.opts.sendBuffer),
 		backoffCfg:      n.mgr.opts.backoff,
-		nodeID:          n.ID(),
+		node:            n,
 		latency:         -1 * time.Second,
 		rand:            rand.New(rand.NewSource(time.Now().UnixNano())),
 		responseRouters: make(map[uint64]responseRouter),
+		connEstablished: false,
 	}
 }
 
 func (c *channel) connect(ctx context.Context, conn *grpc.ClientConn) error {
-	var err error
 	c.parentCtx = ctx
+	go c.sendMsgs()
```

Comment:
It is important to start the goroutine regardless of a successful connection or not. This is because we need to have a listener on the c.sendQ channel. Otherwise, c.enqueue() (used in e.g. QuorumCall and Multicast) will block indefinitely. The reconnection logic is also implemented in c.sendMsgs(). The goroutine will be stopped when the node is closed, preventing leaking goroutines.